Library/Homebrew/cask/lib/hbc/cask_loader.rb

```diff
@@ -31,6 +31,12 @@ def initialize(path)
 
         @token = path.basename(".rb").to_s
         @path = path
+
+        @tap ||= if @path == CaskLoader.default_path(@token)
```

Comment:
What I meant here was that it should only be set by the `FromTapLoader`, since this is the only case where we can be sure of the Tap. And `Cask#full_name` will still default to `Hbc.default_tap` if it is `nil`.